src/model/services/ContactFormService.php

```diff
@@ -0,0 +1,70 @@
+<?php
+require_once 'session_config.php';
+session_start();
+
+class ContactFormService {
+  public function sendMail(): array {
+    $errors = [];
+
+    if (isset($_POST['submit'])) {
+      $formData = [
+        'name' => $_POST['name'],
+        'email' => $_POST['email'],
+        'message' => $_POST['message'],
+      ];
+
+      // Validate the form data inputs
+      $this->validateFormInputs($formData, $errors);
+
+      //Sanitize the form data
+      $formData['name'] = htmlspecialchars(trim($formData['name']));
+      $formData['email'] = htmlspecialchars(trim($formData['email']));
+      $formData['message'] = htmlspecialchars(trim($formData['message']));
+
+      $subject = "Contact Form Submission";
+      $mailTo = "dwp@spicypisces.eu";
+      $headers = "From: " . $formData['email'];
+      $txt = "You have received an email from " . $formData['name'] . ".\n\n" . $formData['message'];
+
+      if(count($errors) == 0) {
+         // Send email and handle success/failure
+         if (mail($mailTo, $subject, $txt, $headers)) {
+          return $errors;
+        } else {
+          $errors['general'] = "Failed to send email.";
+          return $errors;
+        }      
+      } else {
+        return $errors;
+      }
+    }
+  }
+
+  private function validateFormInputs(array $formData, array &$errors): void {
```

Comment:
should we maybe also validate the max length of name and message?

Reply:
yes i think we should